### ISSUE TITLE
syncContextDidSave calling notifyAboutChangedObjectsFromSaveNotification

### ISSUE THREAD

**Issue by bluk** [objcio/core-data#26]
In https://github.com/objcio/core-data/blob/master/Moody/MoodySync/SyncContextOwner.swift#L65 , does anyone know why the `notifyAboutChangedObjectsFromSaveNotification` is called?

The sync context is saved and then the changes are merged into the main managed object context in the `ContextOwnerType`'s `syncContextDidSave` function. It seems that since the changes are coming from the sync context, that we shouldn't have to basically call `performLocalChanges` in `notifyAboutChangedObjectsFromSaveNotification`.

Thanks for any help anyone can provide!

**Comment by timaktimak** [comment]
@bluk Do you mean line 71? I have the same question.

**Comment by epaga** [comment]
Removing this actually fixed something for me - I was syncing to CloudKit and if I was offline, making some changes, then went online, but it re-synced to the old data before I went offline, due to this line. I think it may be a bug.

**Comment by timaktimak** [comment]
@bluk There is a small section in the book explaining how to handle ordering between changeProcessors. So, basically if one of your changeProcessors needs to process objects only after they have been processed by another changeProcessor, you can adjust the predicate so that the objects will satisfy the predicate only after being processed by the first changeProcessor (for example, if you changeProcessor needs to work only with the uploaded objects, your predicate should contain `remoteID != NULL`). Once the first changeProcessor processes the objects, it will then set some of the fields of these objects on the syncContext, and because of the `performLocalChanges` these changed objects will be picked and processed by the second changeProcessor immediately, and not after another trigger of the synchronization.
